### DISCUSSION TITLE
Naive Strategy

### DISCUSSION THREAD

**Discussion by MagicHealer** [ContinualAI/avalanche#1491] [Q&A] [answered]
Can you explain what is actually happening in Naive strategy? Thanks

**Comment by HamedHemati** [comment, marked as answer]
Naive strategy is the most basic type of a continual learner. For every arriving experience, it iterates through the samples of the experience's dataset and updates the model with an SGD-based optimizer without any additional regularization or augmentation methods. It is also called sequential fine-tuning. 

If you are looking for more technical details, an SGD-based template (the super-class of the Naive strategy) in Avalanche provides a sequence of callbacks that are triggered one after another when you call `strategy.train([exp])`. Here is an example at the high level:

 "create dataloader"-> "get batch i" -> "forward" -> "calculate loss" -> "backward" -> etc.

Here is a a plot from https://arxiv.org/pdf/2302.01766.pdf that shows some of the call backs in the Base and SGD templates:

<img width="1205" alt="sgd-template" src="https://github.com/ContinualAI/avalanche/assets/11629528/025dd3c7-00d9-4544-b9e4-303f9d280093">

Link to the API doc and source code:
https://avalanche-api.continualai.org/en/v0.4.0/generated/avalanche.training.Naive.html#avalanche.training.Naive
